File: Problems_and_Solutions/1237_Find-Positive-Integer-Solution-for-a-Given-Equation/Solution.java
/*
 * // This is the custom function interface.
 * // You should not implement it, or speculate about its implementation
 * class CustomFunction {
 *     // Returns f(x, y) for any given positive integers x and y.
 *     // Note that f(x, y) is increasing with respect to both x and y.
 *     // i.e. f(x, y) < f(x + 1, y), f(x, y) < f(x, y + 1)
 *     public int f(int x, int y);
 * };
 */
class Solution {
    public List<List<Integer>> findSolution(CustomFunction customfunction, int z) {
        List<List<Integer>> res = new ArrayList<>();
        int x = 1, y = 1000;
        
        while ( x <= 1000 && y >= 1) {
            int tmp = customfunction.f(x,y);
            if (tmp < z) {
                x++;
            } else if (tmp > z) {
                y--;
            } else {
                res.add(Arrays.asList(x++, y--));
            }
        }
        return res;
    }
}

File: Problems_and_Solutions/1237_Find-Positive-Integer-Solution-for-a-Given-Equation/Solution.py
"""
   This is the custom function interface.
   You should not implement it, or speculate about its implementation
   class CustomFunction:
       # Returns f(x, y) for any given positive integers x and y.
       # Note that f(x, y) is increasing with respect to both x and y.
       # i.e. f(x, y) < f(x + 1, y), f(x, y) < f(x, y + 1)
       def f(self, x, y):
  
"""


class Solution:
    def findSolution(self, customfunction: 'CustomFunction', z: int) -> List[List[int]]:
        res = []
        x, y = 1, 1000
        while x <= 1000 and y >= 1:
            tmp = customfunction.f(x, y)
            if tmp < z:
                x += 1
            elif tmp > z:
                y -= 1
            else:
                res.append([x, y])
                x += 1
                y -= 1
        return res
